source/tank/TankFactory.hx

```diff
@@ -18,4 +22,20 @@ abstract class TankFactory {
 		tank.setControllers(new KeyboardMoveController(tank), new MouseShootController(tank));
 		return tank;
 	}
+
+	static inline public function NewDynamicTank(X:Float = 0, Y:Float = 0, target:Tank,
+			map:FlxTilemap) {
+		var tank = new Tank(X, Y, AssetPaths.green_tank_body__png);
+		tank.setControllers(new DynamicMovementController(tank, target, map),
+			new AutoAimController(tank, target, map));
+		return tank;
+	}
+
+	static inline public function NewPursuitTank(X:Float = 0, Y:Float = 0, target:Tank,
+			map:FlxTilemap) {
+		var tank = new Tank(X, Y, AssetPaths.new_tank_body__png);
+		tank.setControllers(new PursuePlayerMovementController(tank, target),
+			new AutoAimController(tank, target, map));
+		return tank;
+	}
```

Comment:
Good work putting together these factory methods, I especially like the look of the changed tank sprites you picked 👍